herd/constraints.ml

```diff
@@ -120,16 +120,28 @@ module Make (C:Config) (A : Arch_herd.S) :
             let rec do_rec = function
               | Atom (LV (rloc,v)) ->
                  let t = look_type rloc in
-                 let w = look_val rloc
+                 let w = match A.get_tagloc_loc (loc_of_rloc rloc) with
+                   | Some loc ->
+                     let tag = look_val rloc in
+                     A.update_loc_tag (loc) tag
```

Comment:
Rather than creating a new Location_global with default data and setting the tag can't you just compare the tags?